src/gpu/intel/jit/dsl/dsl.cpp

```diff
@@ -30,34 +30,52 @@ namespace jit {
 namespace dsl {
 
 struct ctx_t {
+    bool new_ir_api() const { return new_ir_api_; }
 
-    void declare_kernel(const kernel_iface_t &interface, ir_context_t &ctx) {
+    void declare_kernel(const kernel_iface_t &interface, ir_context_t &ctx,
+            bool new_ir_api = false) {
+        slm_byte_offset_ = 0;
+        new_ir_api_ = new_ir_api;
```

Comment:
Why do we need this? At this point we barely have any compatibility concerns, so I don't see why we would merge compatibility functionality at this point.

Reply:
Well, we do :) #3729 relies on the current semantics.

I have an implementation relying on the new semantics, so my plan is to merge the current work and then follow-up with specific adjustments. We can do all at once but I think it's less productive.